clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -367,9 +369,13 @@ private void checkDisconnects(long now) {
                 Collection<ClientRequest> requests = unsent.remove(node);
                 for (ClientRequest request : requests) {
                     RequestFutureCompletionHandler handler = (RequestFutureCompletionHandler) request.callback();
-                    handler.onComplete(new ClientResponse(request.makeHeader(request.requestBuilder().desiredOrLatestVersion()),
-                        request.callback(), request.destination(), request.createdTimeMs(), now, true,
-                        null, null));
+                    AuthenticationException authenticationException = client.authenticationException(node);
+                    if (authenticationException !=  null)
+                        handler.onFailure(authenticationException);
```

Comment:
I guess it's possible that we could get this error from the consumer's heartbeat thread. If so, we will swallow it. Same thing if we get a `GroupAuthorizationException` I guess. We actually do have a way to propagate exceptions from the heartbeat thread to the `poll()`. Maybe we should try to handle these two cases?

Reply:
I added handlers for these two error cases.